### DISCUSSION TITLE
gdb stub support for esp32

### DISCUSSION THREAD

**Discussion by tt9966** [zephyrproject-rtos/zephyr#70963] [RFC]
I am trying to debug use GDB stub as pin count limitation workaround for a "esp32_devkitc_wrover" compatible board.
Using the extra 4 jtag pins for debugging, really limits the platform connection to outside world, spi etc..

It looks like compile indicate there is missing support in esp32 framwork or is it not expected used in this way?
ESP32 maybe missing [Architecture Porting](https://docs.zephyrproject.org/latest/hardware/porting/arch.html)
```
zephyrproject/zephyr/arch/xtensa/core/gdbstub.c: In function 'copy_to_ctx':

...

zephyrproject/zephyr/arch/xtensa/core/gdbstub.c:449:57:
error: 'BASE_SAVE_AREA_SIZE' undeclared (first use in this function)
  449 |                                         ((char *)bsa) + BASE_SAVE_AREA_SIZE);
      |                                                         ^~~~~~~~~~~~~~~~~~~
```

Is there another way of using only the default usb connected serial for "real" gdb debug?
Using printk is somewhat a limit for me in normal firmware write.

There error looks not to be related compile to a shared "uart". as it did not help disable and enabling normal shell and CONFIG_CONSOLE=n
For future, can ESP32 shell and logging be moved to bluetooth/wifi to free up USB (uart0)?

Compile is done with current released zephyr 3.6.0

[example to demonstrate how GDB stub works](https://docs.zephyrproject.org/latest/services/debugging/gdbstub.html#example)

`west build -p -b esp32_devkitc_wrover samples/subsys/debug/gdbstub`

With overlay in "samples/subsys/debug/gdbstub/boards/esp32_devkitc_wrover.overlay"
```
/ {
	chosen {
		zephyr,gdbstub-uart = &uart0;
	};
};

&uart0{
	status="okay";
};
```

**Comment by rftafas** [comment]
> It looks like compile indicate there is missing support in esp32 framwork or is it not expected used in this way?

Right observation. You would need the [USB-JTAG Serial on-chip controller](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-guides/usb-serial-jtag-console.html) present on ESP32-C3 and newer devices. For ESP32 you need to use the pins.

**Reply by tt9966** [reply]
Thanks for the feed back, was hopping for a dual core feature in zephyr was able to do it in software :-)
